src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -915,5 +918,82 @@ void AppHost::_HandleSettingsChanged(const winrt::Windows::Foundation::IInspecta
 void AppHost::_IsQuakeWindowChanged(const winrt::Windows::Foundation::IInspectable&,
                                     const winrt::Windows::Foundation::IInspectable&)
 {
+    if (_window->IsQuakeWindow() && !_logic.IsQuakeWindow())
+    {
+        // If we're exiting quake mode, we should make our
+        // tray icon disappear.
+        if (_trayIconData)
+        {
+            Shell_NotifyIcon(NIM_DELETE, &_trayIconData.value());
+            _trayIconData.reset();
+        }
+    }
+
     _window->IsQuakeWindow(_logic.IsQuakeWindow());
 }
+
+void AppHost::_HandleTrayIconPressed()
+{
+    // Currently scoping "minimize to tray" to only
+    // the quake window.
+    if (_logic.IsQuakeWindow())
+    {
+        _window->SummonWindow({});
+    }
+}
+
+void AppHost::_HandleWindowHidden()
+{
+    if (_logic.IsQuakeWindow())
+    {
+        _UpdateTrayIcon();
+    }
+}
+
+// Method Description:
+// - Creates and adds an icon to the notification tray.
+// Updates the current tray icon with the current window's
+// HWND if an icon already exists.
+// Arguments:
+// - <unused>
+// Return Value:
+// - <none>
+void AppHost::_UpdateTrayIcon()
+{
+    if (!_trayIconData)
+    {
+        NOTIFYICONDATA nid{};
+
+        // This HWND will receive the callbacks sent by the tray icon.
+        nid.hWnd = _window->GetHandle();
+
+        // App-defined identifier of the icon. The HWND and ID are used
+        // to identify which icon to operate on when calling Shell_NotifyIcon.
+        // Multiple icons can be associated with one HWND, but here we're only
+        // going to be showing one so the ID doesn't really matter.
+        nid.uID = 1;
+
+        nid.uCallbackMessage = WM_APP + 1;

```

Comment:
Unsure what to call this magic message and where to put it 🤔

Reply:
we can probably stick this in `IslandWindow.h`, and it can probably just be `CM_NOTIFY_FROM_TRAY` 

(there are other examples of us using `CM_` naming in `windowproc.cpp`)